pkg/operator/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -81,25 +82,51 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 
 	// only manage the apiservices if we have ready pods for the daemonset.  This makes sure that if we're taking over for
 	// something else, we don't stomp their apiservices until ours have a reasonable chance at working.
+	var actualAPIServices []*apiregistrationv1.APIService
 	if actualDaemonSet != nil && actualDaemonSet.Status.NumberReady > 0 {
-		err = manageAPIServices_v311_00_to_latest(c.apiregistrationv1Client)
+		actualAPIServices, err = manageAPIServices_v311_00_to_latest(c.apiregistrationv1Client)
 		if err != nil {
 			errors = append(errors, fmt.Errorf("%q: %v", "apiservices", err))
 		}
 	}
 
 	// manage status
-	if actualDaemonSet.Status.NumberAvailable > 0 {
+	var availableConditionReason string
+	var availableConditionMessages []string
+	if actualDaemonSet != nil && actualDaemonSet.Status.NumberAvailable == 0 {
+		availableConditionReason = "NoAPIServerNode"
+		availableConditionMessages = append(availableConditionMessages, "no openshift-apiserver daemon pods available on any node.")
+	}
+	for _, apiService := range actualAPIServices {
```

Comment:
If you don't have *all* the APIservices (this can happen on errors), you must indicate that availability is unknown.

Reply:
> If you don't have _all_ the APIservices (this can happen on errors), you must indicate that availability is unknown.

add a test for this.  It's a correctness problem.

Reply:
This is still outstanding.  You would reporting failing and you *also* need to report unavailable.

Reply:
> If you don't have _all_ the APIservices (this can happen on errors), you must indicate that availability is unknown.

* It appears that I either get all `APIServices` , or none if there is **any** error invoking `manageAPIServices_v311_00_to_latest`. 
* When there is an error invoking `manageAPIServices_v311_00_to_latest`, the error will appear in the `WorkloadFailng` condition's message and the `Available` condition would of been set to `Unknown` in the preceding `switch` statement.